src/main.rs

```diff
@@ -152,10 +166,12 @@ async fn upload_project(
     )
     .await?;
 
-    let _ = db.conn().insert_upload(&upload)?;
+    let _ = db.conn().insert_upload(&upload_entry)?;
 
     // Clean up the temporary directory.
-    fs::remove_dir_all(upload_dir).map_err(|_| ApiError::Upload(UploadError::SaveFile))?;
+    tmp_dir
```

Comment:
whole point of tempfile or tempdir is not requiring explicit cleanup since the `Drop` trait for this should do this when the variable goes out of scope (I think)

Reply:
The docs say to call `tmpdir.close()`